File: Structural/Adapter/C#/AdapterExample/AdapterExample/PrinterAdapter.cs
namespace AdapterExample;

public class PrinterAdapter : ITextFormatter
{
	private readonly OldPrinter _printer;

	public PrinterAdapter(OldPrinter printer)
	{
		_printer = printer;
	}

	public void FormatAndPrint(string text)
	{
		var formattedText = "<h1>" + text + "</h1>";
		_printer.PrintLine(formattedText);
	}
}

File: Structural/Adapter/Delphi/PrinterAdapter.pas
unit PrinterAdapter;

interface

uses
  TextFormatter, OldPrinter;

type
  TPrinterAdapter = class(TInterfacedObject, ITextFormatter)
  private
    FPrinter: TOldPrinter;
  public
    constructor Create(Printer: TOldPrinter);
    procedure FormatAndPrint(const text: string);
  end;

implementation

constructor TPrinterAdapter.Create(Printer: TOldPrinter);
begin
  inherited Create();
  FPrinter := Printer;
end;

procedure TPrinterAdapter.FormatAndPrint(const text: string);
var
  formattedText: string;
begin
  formattedText := '<h1>' + text + '</h1>';
  FPrinter.PrintLine(formattedText);
end;


end.
